sidecar/src/controller.rs

```diff
@@ -20,7 +20,7 @@ pub(crate) struct Controller {
     /// Check interval
     check_interval: Interval,
     /// graceful shutdown signal
-    graceful_shutdown: EventListener,
+    graceful_shutdown: Receiver<()>,
```

Comment:
How come you replace `EventListener` with `Receiver`? Is it better?

Reply:
The reason similar to [xline's graceful shutdown](https://github.com/xline-kv/Xline/pull/396#issuecomment-1649766700).
When running multiple tasks in parallel using `tokio::spawn`, it's essential to have a "callback" mechanism to signal the `run` once all tasks have finished.
Implementing it using an `Receiver` is cleaner than `EventListener`.